feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-image/configurator-attribute-multi-selection-image.component.ts

```diff
@@ -1,33 +1,27 @@
-import {
-  ChangeDetectionStrategy,
-  Component,
-  EventEmitter,
-  Input,
-  OnInit,
-  Output,
-} from '@angular/core';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
 import { FormControl } from '@angular/forms';
-import { Configurator } from '../../../../core/model/configurator.model';
-import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
 import { ConfiguratorStorefrontUtilsService } from '../../../service/configurator-storefront-utils.service';
-import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorAttributeMultiSelectionBaseComponent } from '../base/configurator-attribute-multi-selection-base.component';
+import { ConfiguratorUiKeyGeneratorService } from '../base/configurator-ui-key-generator.service';
+
 @Component({
   selector: 'cx-configurator-attribute-multi-selection-image',
   templateUrl: './configurator-attribute-multi-selection-image.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ConfiguratorAttributeMultiSelectionImageComponent
-  extends ConfiguratorAttributeBaseComponent
+  extends ConfiguratorAttributeMultiSelectionBaseComponent
   implements OnInit {
-  @Input() attribute: Configurator.Attribute;
-  @Input() ownerKey: string;
-
-  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
-
   constructor(
-    protected configUtilsService: ConfiguratorStorefrontUtilsService
+    /**
+     * @deprecated since 4.1: remove ConfiguratorStorefrontUtilsService dependency
+     */
+    protected configUtilsService: ConfiguratorStorefrontUtilsService,
+    protected quantityService: ConfiguratorAttributeQuantityService,
+    protected uiKeyGeneratorService: ConfiguratorUiKeyGeneratorService
   ) {
-    super();
+    super(quantityService, uiKeyGeneratorService);
```

Comment:
we can use the parent's constructor, if we don't consider breaking changes :)

Reply:
Done!